### ISSUE TITLE
DeprecationWarning with Python 3.7

### ISSUE THREAD

**Issue by dmajda** [aio-libs/aiokafka#509]
When using aiokafka with Python 3.7 with warnings enabled, the following deprecation warning is emitted:

```
/usr/local/lib/python3.7/dist-packages/aiokafka/client.py:417: DeprecationWarning: 'with (yield from lock)' is deprecated use 'async with lock' instead
  with (yield from self._get_conn_lock):
```

It would be good if aiokafka was warning-free on Python 3.7. (In our specific case, this warning pollutes our test suite output.)

The cleanest solution is to rewrite the code to use `async with lock`, but that means dropping Python 3.4 support. In case this is undesirable, another possible fix is to avoid the `with` statement entirely and just use `self._get_conn_lock.acquire()` and `self._get_conn_lock.release()` in a `try`/`finally` block. Not sure if there is any other solution.

In any case, if any of the suggested solutions is acceptable, I can prepare a patch.

### Steps to reproduce

Assuming that Kafka is running on `localhost:9092`.

```console
$ python3.7 --version
Python 3.7.1
$ python3.7 -m pip install aiokafka
[...]
$ cat test.py
import aiokafka
import asyncio

async def main():
    producer = aiokafka.AIOKafkaProducer(loop=asyncio.get_running_loop())
    await producer.start()
    await producer.stop()

asyncio.run(main())
$ python3.7 -Wall test.py
/usr/local/lib/python3.7/dist-packages/aiokafka/client.py:417: DeprecationWarning: 'with (yield from lock)' is deprecated use 'async with lock' instead
  with (yield from self._get_conn_lock):
```

**Comment by asvetlov** [comment]
I think rewriting to async/await is a good way.
But here @tvoinarovskyi decides.
BTW, in python 3.8 `@coroutine` will be deprecated too: https://github.com/python/cpython/pull/13346

**Comment by tvoinarovskyi** [comment]
@asvetlov Yea, I think it's been too long to postpone that work. Will remove support for <3.6 with 0.6.0.

**Comment by tvoinarovskyi** [comment]
Merged, now the upstream should not produce those warnings

**Comment by tvoinarovskyi** [comment]
For now 3.5.3+ pythons are working. @asvetlov do you still support 3.5 for aiohttp? I have a feeling I want to go with 3.6 support because 3.5 is not very consistent on API in asyncio itself, but wandering if that is a valid enough reason.

**Comment by dmajda** [comment]
> Merged, now the upstream should not produce those warnings

Thanks a lot!

**Comment by tvoinarovskyi** [comment]
Should be fixed with release 0.6.0. Feel free to ask if you have issues still.

**Comment by dmajda** [comment]
Thanks, will try after our next dependencies upgrade round. Great to have a new release!
